src/docs/development/platform-integration/add-to-app-android/add-flutter-fragment.md

```diff
@@ -0,0 +1,311 @@
+        // this example, assume that a FrameLayout exists with an ID of 
+        // R.id.fragment_container.
+        setContentView(R.layout.my_activity_layout);
+
+        // Get a reference to the Activity's FragmentManager to add a new 
+        // FlutterFragment, or find an existing one.
+        FragmentManager fragmentManager = getSupportFragmentManager();
+
+        // Attempt to find an existing FlutterFragment, in case this is not the
+        // first time that onCreate() has run.
+        flutterFragment = (FlutterFragment) fragmentManager
+            .findFragmentByTag(TAG_FLUTTER_FRAGMENT);
+
+        // Create and attach a FlutterFragment if one does not yet exist.
+        if (flutterFragment == null) {
+            flutterFragment = new FlutterFragment.Builder().build();
+            
+            fragmentManager
+                .beginTransaction()
+                .add(
+                    R.id.fragment_container, 
+                    flutterFragment, 
+                    TAG_FLUTTER_FRAGMENT
+                )
+                .commit();
+        }
+    }
+}
+```
+
+The above code is sufficient to render a Flutter UI that begins with a call to
+your `main()` Dart entrypoint and an initial Flutter route of `/`. However, this
+code is not sufficient to achieve all expected Flutter behavior. Flutter depends
+on various OS signals that need to be forwarded from your host `Activity` to
+`FlutterFragment`. These calls are shown below:
+
+```java
+public class MyActivity extends FragmentActivity {
+    @Override
+    public void onPostResume() {
+        super.onPostResume();
+        flutterFragment.onPostResume();
+    }
+
+    @Override
+    protected void onNewIntent(@NonNull Intent intent) {
+        flutterFragment.onNewIntent(intent);
+    }
+
+    @Override
+    public void onBackPressed() {
+        flutterFragment.onBackPressed();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(
+        int requestCode, 
+        @NonNull String[] permissions, 
+        @NonNull int[] grantResults
+    ) {
+        flutterFragment.onRequestPermissionsResult(
+            requestCode, 
+            permissions, 
+            grantResults
+        );
+    }
+
+    @Override
+    public void onUserLeaveHint() {
+        flutterFragment.onUserLeaveHint();
+    }
+
+    @Override
+    public void onTrimMemory(int level) {
+        super.onTrimMemory(level);
+        flutterFragment.onTrimMemory(level);
+    }
+}
+```
+
+With the above OS signals forwarded to Flutter, your `FlutterFragment` works as
+expected. You have now added a `FlutterFragment` to your existing Android app.
+
+## Using a pre-warmed `FlutterEngine`
+
+By default, a `FlutterFragment` creates its own instance of a `FlutterEngine`,
+which requires non-trivial warmup time. This means your user sees a blank
+`Fragment` for a second or two. You can mitigate most of this warmup time by
```

Comment:
A second or 2 is probably a bit pessimistic. Even our flutter_gallery__start_up μstimeToFirstFrameMicros only takes 750ms and that includes the application launch. We don't have to give numbers here since we'll likely have a dedicated page for this.

Reply:
Updated to say a "moment"